Some Dockerfile changes and README fixes

AdminRAT opened the pull request:
I have made some changes in the Dockerfile and Docker-Compose files that would save a lot of space (964 MB down to 121 MB) and that you should not build an image before running `docker-compose up`. I even added some comments in the example config that would make it easier for one to self-host the bot themselves. And lastly, fix up the README a bit.

Commit:
Use python 3.7 alpine image instead of standard

Commit:
Use build instead of image in docker compose

Commit:
Fix up README and add example.env

Commit:
Use /app instead of /blurple in Dockerfile
I forgot that in the docker compose config there is a volume to /app

Commit:
Change README from rst to md

Commit:
Update example config and fix an error

Commit:
Merge branch 'master' of https://github.com/project-blurple/blurplefier

Commit:
Pull fork from GitLab
https://gitlab.com/AdminRAT/blurplefier

LostLuma commented on Dockerfile:
```diff
@@ -1,6 +1,12 @@
-FROM python:3.7
+FROM python:3.7-alpine AS requirements
```
Changing to alpine doesn't make a lot of sense for us as the Docker setup is mostly meant for development where other Python based images are likely present, meaning no space is saved while adding additional build time.

AdminRAT replied:
>the Docker setup is mostly meant for development

Ah sorry for the confusion, I initially thought it was ready for self-hosting at the time

>no space is saved  

During the first time I compared the two images, it went from 964 MB to 121 MB

>while adding additional build time

I still didn't realize that alpine does that when building the image

LostLuma commented on Dockerfile:
```diff
@@ -1,6 +1,12 @@
+RUN pip install -r requirements.txt
 
+FROM python:3.7-alpine
+RUN apk add --no-cache jpeg-dev libwebp libwebp-dev
```
This is missing bash as a dependency for the Jishaku bot debugging tools.

AdminRAT replied:
I'll add that

LostLuma commented on docker-compose.yaml:
```diff
@@ -2,7 +2,7 @@ version: '3'
 services:
   bot:
-    image: blurplefy
+    build: .
```
Changing these from using the tag to `build` leads to docker-compose building the exact same image twice, doubling rebuild time.

AdminRAT replied:
Strange, when I was initially building it using docker-compose it would re-use the cache from the first image it built (in this case, the worker).

![image](https://user-images.githubusercontent.com/24538037/57741417-941fba00-7671-11e9-80a4-efb5ae3d43d5.png)

LostLuma commented on Dockerfile:
```diff
@@ -1,6 +1,12 @@
-
-RUN pip install -r requirements.txt
+COPY --from=requirements /usr/local/lib/python3.7/site-packages /usr/local/lib/python3.7/site-packages
+RUN adduser -D -h /app -s /bin/sh -u 1000 blurple
```
Using a different user here makes it difficult to add additional dependencies for testing during development.

LostLuma commented on example-config.yaml:
```diff
@@ -1,7 +1,15 @@
+# You will need to use a custom emoji here for when the bot is processing the image.
+# Get rid of the "a" if you are not using an animated emoji
```
Using unicode emoji is also possible.

AdminRAT replied:
I just realized that. I'll edit the comment then.

LostLuma commented on example-config.yaml:
```diff
@@ -1,7 +1,15 @@
+# To set this up:
+# 1. Create 5 roles
+# 2. Make them mentionable temporarily
+# 3. Grab their ID's (put a \ before the @ when mentioning the role.)
```
Role IDs can now be obtained by right clicking on roles in the role menu or in a users profile.

AdminRAT replied:
I didn't know that until now, I'll add that as well.

LostLuma commented on example-config.yaml:
```diff
@@ -10,22 +18,29 @@ blurple_dark_role: 573651772812558337
   - second.worker.token
   # Add as many as needed
 
+# Leave this alone since this is already configured for docker. (Unless you are setting it up on Windows, change the address)
```
Docker and docker-compose work just fine on windows, if you want to help users setting it up without Docker there should be an OS agnostic tip to set it to 127.0.0.1 or localhost.

AdminRAT replied:
I'll edit that as well.

Commit:
Pull more changes from GitLab
I hope this is better if anything...
https://gitlab.com/AdminRAT/blurplefier on development branch
